installation/developer-installation/full-developer-installation.md

```diff
@@ -0,0 +1,247 @@
+
+    OctoBot's multiple repos, this is just a quality of life
+
+    improvement and is not necessary.
+* Language: [Python 3.8](https://www.python.org/downloads/)
+
+#### Dependencies for Windows
+
+* download [Visual Studio build tools](https://visualstudio.microsoft.com/downloads/#build-tools-for-visual-studio-2019) and install "Desktop development with C++"
+
+![](<../../.gitbook/assets/image (6).png>)
+
+![](<../../.gitbook/assets/image (1).png>)
+
+#### Dependencies for Mac
+
+* install [GCC](https://discussions.apple.com/thread/8336714)
+
+## Cloning OctoBot repositories
+
+* Clone each OctoBot repository using the dev branch when specified.
+
+### Cloning script for unix setups
+
+A script to install these git repos on a unix setup
+
+```bash
+#!/bin/bash
+
+readonly REMOTE_DEVBRANCH="remotes/origin/dev"
+readonly DEVBRANCH="dev"
+readonly BASEDIR=$(dirname "$0")
+
+branch_work() {
+  dir=$1
+  devbranch=$2
+  branch=$(cd $BASEDIR/$dir && git name-rev --name-only HEAD)
+  if $devbranch; then
+    if [ $branch == $REMOTE_DEVBRANCH ]; then
+      echo "[WARN] Already on branch: $branch "
+      echo "[INFO] Delete Folder: $dir - if you would like to have clean $dir project"
+    else
+      (cd $dir; git checkout $DEVBRANCH)
+    fi
+  fi
+}
+
+project_work() {
+  url=$1
+  devbranch=$2
+  dir=$(basename -s .git "$url")
+  echo "Check Dir $BASEDIR/$dir"
+  if [ -d $BASEDIR/$dir ]; then
+    echo "[WARN] Directory: $dir exists!"
+    branch_work $dir $devbranch
+  else
+    echo "----- $dir -----"
+    git clone $url
+    branch_work $dir $devbranch
+    echo "----- END $dir -----"
+  fi
+}
+
+#Uses dev branch: true/false
+project_work https://github.com/Drakkar-Software/OctoBot.git true
+project_work https://github.com/Drakkar-Software/OctoBot-Tentacles.git true
+project_work https://github.com/Drakkar-Software/OctoBot-Trading false
+project_work https://github.com/Drakkar-Software/OctoBot-evaluators false
+project_work https://github.com/Drakkar-Software/OctoBot-Services false
+project_work https://github.com/Drakkar-Software/OctoBot-Backtesting false
+project_work https://github.com/Drakkar-Software/OctoBot-Tentacles-Manager false
+project_work https://github.com/Drakkar-Software/OctoBot-Commons false
+project_work https://github.com/Drakkar-Software/Async-Channel false
+```
+
+### Cloning OctoBot repositories with GitKraken
+
+* open up GitKraken and press on Clone a repo
+
+![](<../../.gitbook/assets/image (4).png>)
+
+*   Clone the repository into the preferred destination
+
+    and repeat that step for all repositories [below](full-developer-installation.md#octobot-repositories)
+
+![](<../../.gitbook/assets/image (18).png>)
+
+* now you should have all the Octobot modules in one folder
+
+![](<../../.gitbook/assets/image (8).png>)
+
+#### OctoBot repositories
+
+OctoBot code is split in multiple repositories:
+
+*   [https://github.com/Drakkar-Software/OctoBot](https://github.com/Drakkar-Software/OctoBot) (dev branch) for the
```

Comment:
in this listing, we should remove all these newlines inside items, it creates weird line breaks